src/main/java/mezz/jei/gui/recipes/OutputSlotTooltipCallback.java

```diff
@@ -0,0 +1,69 @@
+package mezz.jei.gui.recipes;
+
+import mezz.jei.api.gui.ingredient.IRecipeSlotTooltipCallback;
+import mezz.jei.api.gui.ingredient.IRecipeSlotView;
+import mezz.jei.api.helpers.IModIdHelper;
+import mezz.jei.api.ingredients.IIngredientHelper;
+import mezz.jei.api.ingredients.ITypedIngredient;
+import mezz.jei.api.recipe.RecipeIngredientRole;
+import mezz.jei.api.runtime.IIngredientManager;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.List;
+import java.util.Optional;
+
+public class OutputSlotTooltipCallback implements IRecipeSlotTooltipCallback {
+	private final ResourceLocation recipeName;
+	private final IModIdHelper modIdHelper;
+	private final IIngredientManager ingredientManager;
+
+	public OutputSlotTooltipCallback(
+		ResourceLocation recipeName,
+		IModIdHelper modIdHelper,
+		IIngredientManager ingredientManager
+	) {
+		this.recipeName = recipeName;
+		this.modIdHelper = modIdHelper;
+		this.ingredientManager = ingredientManager;
+	}
+
+	@Override
+	public void onTooltip(IRecipeSlotView recipeSlotView, List<Component> tooltip) {
+		Optional<ITypedIngredient<?>> displayedIngredient = recipeSlotView.getDisplayedIngredient();
+		if (displayedIngredient.isEmpty()) {
+			return;
+		}
+		if (recipeSlotView.getRole() != RecipeIngredientRole.OUTPUT) {
+			return;
+		}
```

Comment:
I feel like this check potentially should be done before checking if the displayed ingredient is empty as it is a more straightforward check and if it isn't an output there is no reason to bother checking which ingredient is currently displayed if any.